src/hooks/useUpdate.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+const useUpdate = ({ action, exit }, ...args) => {
```

Comment:
Can we instead do something like this? 🤔 

```js
const useUpdate = ({ action = () => {}, exit = () => {} }, ...args) => {
  React.useEffect(() => {
    if (action && typeof action === 'function') {
      params.action();
    }
    if (exit && typeof exit === 'function') {
      return params.exit();
    }
  }, args.length ? [...args] : undefined);
};
```

@umairfarooq44 what do you think?

Reply:
```diff
-  }, args.length ? [...args] : undefined);
+ }, args.length ? [...args] : [true]); // probably have to experiment with both of them to see which works 💭 
```

Reply:
On further deep diving into the react's code base the second argument if `undefined` will work as the default second argument to `React.useEffect()`

Ref: https://github.com/facebook/react/blob/master/packages/react/src/ReactHooks.js#L87